### PR TITLE
Add heist and heist-salt runner

### PR DESCRIPTION
**Pull request by Ch3LL**
### What does this PR do?
Adds the heist and heist-salt runner.

### CODE REVIEW

File: changelog/60556.added [hunk 1]
```diff
@@ -0,0 +1 @@
+Add the heist and heist-salt runner.
```
Comment (garethgreenaway) [inline changelog/60556.added:1]: Super minor but is the `the` in the wrong place?

**Review by garethgreenaway — APPROVED** [review]

File: tests/pytests/unit/runners/test_heist_salt.py [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+def configure_loader_modules():
+    return {
+        heist: {"__utils__": {"hub.hub": hub.hub}},
+        hub: {"__context__": {}},
+        heist_salt: {"__salt__": {"heist.deploy": heist.deploy}},
+    }
+
+
+@pytest.fixture()
+def test_hub():
```
Comment (s0undt3ch) [inline tests/pytests/unit/runners/test_heist_salt.py:25]: Same as the other test module

File: tests/pytests/unit/utils/test_hub.py [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+pytest.importorskip("heist", reason="Test requires heist to be installed")
+
+
+@pytest.fixture()
+def configure_loader_modules():
+    return {hub: {"__context__": {}}}
+
+
+@pytest.fixture()
+def test_hub():
```
Comment (s0undt3ch) [inline tests/pytests/unit/utils/test_hub.py:14]: Same as the other test modules

**Review by s0undt3ch — CHANGES_REQUESTED** [review]

**Comment by Ch3LL** [comment]
closing as i will work on migrating this to a salt-extension project.

**Pull request closed without merging**